src/OpenFeature/EventExecutor.cs

```diff
@@ -0,0 +1,294 @@
+                    Type = eventType,
+                    Message = message
+                });
+            }
+        }
+
+        private async Task ProcessFeatureProviderEventsAsync(FeatureProviderReference providerRef)
+        {
+            while (true)
+            {
+                var item = await providerRef.Provider.GetEventChannel().Reader.ReadAsync().ConfigureAwait(false);
+
+                switch (item)
+                {
+                    case ProviderEventPayload eventPayload:
+                        await this.EventChannel.Writer.WriteAsync(new Event { Provider = providerRef, EventPayload = eventPayload }).ConfigureAwait(false);
+                        break;
+                    case ShutdownSignal _:
+                        providerRef.ShutdownSemaphore.Release();
+                        return;
+                }
+            }
+        }
+
+        // Method to process events
+        private async Task ProcessEventAsync()
+        {
+            while (true)
+            {
+                var item = await this.EventChannel.Reader.ReadAsync().ConfigureAwait(false);
+
+                switch (item)
+                {
+                    case Event e:
+                        this._mutex.WaitOne();
+                        if (this._apiHandlers.TryGetValue(e.EventPayload.Type, out var eventHandlers))
+                        {
+                            foreach (var eventHandler in eventHandlers)
+                            {
+                                eventHandler.Invoke(e.EventPayload);
+                            }
+                        }
+
+                        // look for client handlers and call invoke method there
+                        foreach (var keyAndValue in this._namedProviderReferences)
+                        {
+                            if (keyAndValue.Value == e.Provider)
+                            {
+                                if (this._clientHandlers.TryGetValue(keyAndValue.Key, out var clientRegistry))
+                                {
+                                    if (clientRegistry.TryGetValue(e.EventPayload.Type, out var clientEventHandlers))
+                                    {
+                                        foreach (var eventHandler in clientEventHandlers)
+                                        {
+                                            eventHandler.Invoke(e.EventPayload);
+                                        }
+                                    }
+                                }
+                            }
+                        }
+                        this._mutex.ReleaseMutex();
+                        break;
+                    case ShutdownSignal _:
+                        this._shutdownSemaphore.Release();
+                        return;
+                }
+
+            }
+        }
+
+        // Method to signal shutdown
+        public async Task SignalShutdownAsync()
```

Comment:
I assume this needs to be wired to OpenFeature shutdown ? (refer - https://github.com/open-feature/dotnet-sdk/blob/2e1b27fdcd0b2dcc1c6e83a7f77eedbef911915a/src/OpenFeature/Api.cs#L206-L209) 🤔